src/IW/Core/Repo.hs

```diff
@@ -1,7 +1,19 @@
+{-# LANGUAGE DeriveAnyClass #-}
+
 module IW.Core.Repo
-       ( Repo
+       ( Repo (..)
        ) where
+
+import IW.Core.Id (Id (..))
+import IW.Core.SqlArray (SqlArray (..))
         
 
--- | Data type representing a GitHub repository.
-data Repo
+-- | Data type representing a GitHub repository
+data Repo = Repo 
+    { repoId         :: Id Repo
+    , repoOwner      :: Text
+    , repoName       :: Text
```

Comment:
Owner and Repo name are used already in two data types. And they also used as foreign keys. So it's better at this stage to convert `Text` to newtypes. Let's introduce more newtypes inside this module:

1. `RepoOwner`
2. `RepoName`

And we can also use them in the `Issue` data type.

Reply:
```
-- | Unsafe cast of 'Id'. Implementation uses smart trick to enforce usage
-- always with @TypeApplications@.
castId :: forall to from to' . ((to == to') ~ 'True) => Id from -> Id to'
castId (Id a) = Id a
```
Will `Owner` and `Name` need this cast function as well? 
Also, I will add these newtypes to the `Issue` fields in another PR after the other branches related to `Issue` are merged

Reply:
@rashadg1030 These newtypes don't have phantom type parameters. In our system `Owner` can be only for `Repo` so let's have just `newtype RepoOwner` for simplicity. Same for the `Name`. We won't have a user data type for quite a long time. so we will have only `RepoName` We can refactor data types easily later after we have more types and we can notice the pattern. But for now let's just have simple types so type system won't get in the way.